hashtagsv2/graphs/views.py

```diff
@@ -67,9 +67,15 @@ def time_statistics_data(request):
     date_range = latest_date - earliest_date
     # key value pairs of date unit and number of edits
     time_dic = {}
+    
+    if "view_type" in request_dict:
+        view_type = request_dict['view_type']
+    else:
+        view_type = 'NOTA'
 
     # Split by days
-    if date_range.days < 90:
+    if (date_range.days < 90 or view_type == 'daily') and view_type != 'monthly':
```

Comment:
This code was a little confusing to read. I see that it works, it was just a little complicated to parse because we're doing `if view_type != 'monthly' ` here, then `if view_type != 'yearly'` next, then defaulting to years. Is `and view_type != 'monthly'` necessary? If the `view_type` isn't 'daily' then we'll go on to the next entry anyway, right? Or am I misunderstanding?

Reply:
To use the same if-else statements for default output and user requested output, this has to be included. When the API is called for the first time on loading the page, default output will be selected on the basis of number of days.(in this case view_type is defined to prevent any error from using undefined view_type.)
```
    else:
        view_type = 'NOTA'
```

But for the subsequent times, when the API is called by user clicking buttons, it is sending the view_type in get request as a param. So this time the output should be based only on view_type not on number of days(default one). So for example if a 'monthly' view_type is requested for a hashtag with day range for 70 days, it will execute the first if statement on the basis of `date_range.days < 90`, so as to prevent this it is checking that it didn't requested 'monthly' view_type , I set an **and** condition  `view_type!='monthly`.( same reason for `view_type!='yearly'` in second elif statement.)

But as you pointed out this seems to confusing to read. The alternate can be assigning number_of_days a large value so it doesn't affect the result of else-if.

Reply:
Your alternate method looks good and readable now.